soroban-env-common/src/bytes.rs

```diff
@@ -8,14 +10,17 @@ impl<E: Env, const N: usize> TryFromVal<E, BytesObject> for [u8; N] {
     type Error = Error;
 
     fn try_from_val(env: &E, val: &BytesObject) -> Result<Self, Self::Error> {
-        let len: u32 = env.bytes_len(*val).map_err(|_| ConversionError)?.into();
+        let len: u32 = env.bytes_len(*val).map_err(Into::into)?.into();
```

Comment:
Is this changing the error being mapped to? I can't tell what `Into::into` is converting to.

Reply:
yes, it's converting to `Error` such that if we get a `HostError` saying something like "you ran out of gas" we're going to .into() it into the `Error` carried inside that `HostError`, rather than downgrading it to `ConversionError`.